osu.Game/Screens/Backgrounds/BackgroundScreenBeatmap.cs

```diff
@@ -177,9 +179,15 @@ protected override void LoadComplete()
                 BlurAmount.ValueChanged += _ => UpdateVisuals();
             }
 
+            /// <summary>
+            /// Whether the storyboard replace the background
+            /// </summary>
+            private bool showReplaceBackgroundStoryboard => (!ShowStoryboardConfig.Value && !IgnoreUserSettings.Value) || !StoryboardReplacesBackground.Value;
+
             protected override bool ShowDimContent
                 // The background needs to be hidden in the case of it being replaced by the storyboard
-                => (!ShowStoryboard.Value && !IgnoreUserSettings.Value) || !StoryboardReplacesBackground.Value;
+                // But it should show when dim set to 100% anyway to avoid showing Scaling background.
+                => showReplaceBackgroundStoryboard || base.DimLevel >= 1f;
```

Comment:
according to #22238, this is not only cause when scale settings set to 'Everything' and 'excluding overlay', but also multi specter, so it can't just check scale settings.

for dim override, when dim set to 100% or turn into break time, it can cause particularly strange visual effects (backgroundBeatmap show undimmed background while storyboard fades out). So I force it to 100% when storyboardReplaceBackground is enabled